x-pack/platform/plugins/shared/ai_infra/product_doc_base/server/tasks/ensure_up_to_date.ts

```diff
@@ -44,15 +45,17 @@ export const registerEnsureUpToDateTaskDefinition = ({
 export const scheduleEnsureUpToDateTask = async ({
   taskManager,
   logger,
+  inferenceId,
 }: {
   taskManager: TaskManagerStartContract;
   logger: Logger;
+  inferenceId?: string;
 }) => {
   try {
     await taskManager.ensureScheduled({
       id: ENSURE_DOC_UP_TO_DATE_TASK_ID,
       taskType: ENSURE_DOC_UP_TO_DATE_TASK_TYPE,
-      params: {},
+      params: { inferenceId },
```

Comment:
AFAIK `ensureScheduled` tasks are uniquely scheduled by ID. Meaning that if you have different sources installing for different `inferenceId` at the same time, only one will be scheduled. Given security solution and the o11y assistant may call it at the same time at startup, this may be an issue